### ISSUE TITLE
Semi-supervised Multitask Learning for Sequence Labeling

### ISSUE THREAD

**Issue by howardyclo** [howardyclo/papernotes#3]
### Metadata
- Authors: Marek Rei
- Organization: University of Cambridge
- Conference: ACL 2017
- Link: https://goo.gl/h6p29c

**Comment by howardyclo** [comment]
### Summary
This paper propose a auxiliary bidirectional language modeling objective for neural sequence labeling, and evaluated on error detection in learner texts, name entity recognition (NER), chunking and part-of-speech (POS) tagging. 

![Figure 1](https://github.com/howardyclo/papernotes/blob/master/images/Semi-supervised%20Multitask%20Learning%20for%20Sequence%20Labeling/Figure%201.png?raw=true)

Notes on the name of paper:
- "Semi-supervised": Utilize the data without label, which is context words.
- "Multitask Learning": Jointly training on labeling and predicting context words.

---

### Neural Sequence Labeling Model
- Bidirectional LSTM (dynamic weighting char+word embeddings)
- Concatenating forward and backward hidden state as output hidden state at each time step.
- Project each output hidden state to 1-layer feedforward net with *tanh* activation + softmax (or + CRF).
- Code: https://github.com/marekrei/sequence-labeler
---

### Language Modeling Objective
Since bidirectional LSTM has access to the full context on each side of the target token, they **predict the next word only from the forward-moving hidden state** and **the previous word only from the backward-moving hidden state**. (The hidden states are mapped to 1-layer, tanh projection before projecting to context word vocabulary using softmax)

---

### Evaluation
- Evaluated on 10 different sequence labeling datasets.
- Word embeddings are initialized with 300-D pretrained word2vec on Google News for general domains / 200-D pretrained word2vec on PubMed and PMC for biomedical domain.
